modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableView.java

```diff
@@ -2558,9 +2558,10 @@ private void updateTreeEventListener(TreeItem<S> oldRoot, TreeItem<S> newRoot) {
                         int length = e.getTo() - e.getFrom();
                         HashMap<Integer, Integer> pMap = new HashMap<> (length);
                         for (int i = e.getFrom(); i < e.getTo(); i++) {
-                            pMap.put(i, e.getChange().getPermutation(i));
+                            if (e.getChange().wasPermutated()) {
+                                pMap.put(i, e.getChange().getPermutation(i));
+                            }
```

Comment:
Hi Ajit, I am working on fix for [JDK-8193800](https://bugs.openjdk.java.net/browse/JDK-8193800),  This code of handling permutation of table rows is not correct, and as part of the changes for this fix, this code would be changed such that `pMap` will be removed.
Could you please hold this change until [JDK-8193800](https://bugs.openjdk.java.net/browse/JDK-8193800) is fixed.?

Reply:
OK. I will make it a draft PR.
Once you fix JDK-8193800, If no test issue is seen, I will just close this PR.
You may still want to add the UncaughtExceptionHandler in TreeTableViewTest as part of your fix to uncover any similar issue.